tests/error/poly_errors/arg_mismatch4.err

```diff
@@ -0,0 +1,6 @@
+Guppy compilation failed. Error in file $FILE:9
+
+7:    @guppy(module)
+8:    def main(x: array[int, bool]) -> None:
+                  ^^^^^^^^^^^^^^^^
+GuppyTypeError: Expected argument of type `nat`, got type
```

Comment:
The unspecific error location is expected, a bigger refactor is required to track the source locations of type args